src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -60,139 +64,156 @@ CHIP_ERROR OperationalStateServer::Init()
+    if (mOperationalError.errorStateID != to_underlying(ErrorStateEnum::kNoError))
     {
-        err.Set(to_underlying(ErrorStateEnum::kCommandInvalidInState));
+        mOperationalError.Set(to_underlying(ErrorStateEnum::kNoError));
+        ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::OperationalError::Id);
+        MatterReportingAttributeChangeCallback(path);
     }
-    else if (opState != to_underlying(OperationalStateEnum::kPaused))
+
+    uint8_t oldState  = mOperationalState;
+    mOperationalState = aOpState;
+    if (mOperationalState != oldState)
     {
-        delegate->HandlePauseStateCallback(err);
+        ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::OperationalState::Id);
+        MatterReportingAttributeChangeCallback(path);
     }
-    response.commandResponseState = err;
+    return CHIP_NO_ERROR;
+}
 
-    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+DataModel::Nullable<uint8_t> Instance::GetCurrentPhase() const
+{
+    return mCurrentPhase;
 }
 
-void OperationalStateServer::HandleResumeState(HandlerContext & ctx, const Commands::Resume::DecodableType & req)
+uint8_t Instance::GetCurrentOperationalState() const
 {
-    ChipLogDetail(Zcl, "OperationalState: HandleResumeState");
-    Commands::OperationalCommandResponse::Type response;
-    Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
-    GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+    return mOperationalState;
+}
 
-    VerifyOrReturn(delegate != nullptr, ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure));
-    uint8_t opState = delegate->GetCurrentOperationalState();
+void Instance::GetCurrentOperationalError(GenericOperationalError & error) const
+{
+    error.Set(mOperationalError.errorStateID, mOperationalError.errorStateLabel, mOperationalError.errorStateDetails);
+}
 
-    if (opState != to_underlying(OperationalStateEnum::kPaused) && opState != to_underlying(OperationalStateEnum::kRunning))
+void Instance::OnOperationalErrorDetected(const Structs::ErrorStateStruct::Type & aError)
+{
+    ChipLogDetail(Zcl, "OperationalStateServer: OnOperationalErrorDetected");
+    // Set the OperationalState attribute to Error
+    if (mOperationalState != to_underlying(OperationalStateEnum::kError))
     {
-        err.Set(to_underlying(ErrorStateEnum::kCommandInvalidInState));
+        mOperationalState          = to_underlying(OperationalStateEnum::kError);
+        ConcreteAttributePath path = ConcreteAttributePath(mEndpointId, mClusterId, Attributes::OperationalState::Id);
+        MatterReportingAttributeChangeCallback(path);
     }
-    else if (opState == to_underlying(OperationalStateEnum::kPaused))
+
+    // Set the OperationalError attribute
+    if (mOperationalError.IsEqual(aError))
```

Comment:
This check is backward, so this code does not work correctly.  Was this tested?  Why is it not being tested as part of this PR?